src/nupic/algorithms/KNNClassifier.py

```diff
@@ -301,6 +310,11 @@ def removeCategory(self, categoryToRemove):
 
 
   def _removeRows(self, rowsToRemove):
+    """
+    A list of row indices to remove. There are two caveats. First, this is
+    a potentially slow operation. Second, pattern indices will shift if
```

Comment:
The new method is an order of magnitude faster for normal inference and learning. It might be slower for removeRows (didn't check) but only if you remove one row at a time. If you occasionally delete a list of rows, there should be negligible impact.